### PR TITLE
fix(windows): fix command bug in update elevated task (fix #11216)

### LINKED ISSUE #11216: [bug] elevated update task failing when project name has spaces
### Describe the bug

**Platform**: Windows
**Bundle**: Wix

Elevated update task does not take into consideration any spaces that might be present in the path. 
Following command is being run by the [update task](https://github.com/tauri-apps/tauri/blob/dev/crates/tauri-bundler/src/bundle/windows/msi/update-task.xml) 
```xml
<Exec>
      <Command>cmd.exe</Command>
      <Arguments>/c "%SYSTEMROOT%\System32\msiexec.exe /i %TEMP%\\{{product_name}}.msi {{msiexec_args}} /promptrestart"</Arguments>
</Exec>
```

If either the `{{product_name}}` or Username (part of `%TEMP%`) contain spaces, this command will not work. 

Solution:
Adding some escape characters should do the trick.

```xml
<Exec>
      <Command>cmd.exe</Command>
      <Arguments>/c ^"%SYSTEMROOT%\System32\msiexec.exe /i "%TEMP%\\{{product_name}}.msi" {{msiexec_args}} /promptrestart^"</Arguments>
</Exec>
```

### Reproduction

1. Create tauri project with `productName` in `tauri.conf.json` =  `Spaced Project`
2. set `[enableElevatedUpdateTask](https://v2.tauri.app/reference/config/#enableelevatedupdatetask)` to `true`)
3. run cargo tauri build.
4. try to perform the update. 

the application closes but no update happens

### Expected behavior

Update installs

### Full `tauri info` output

```text
[✔] Environment
    - OS: Windows 10.0.26100 x86_64 (X64)
    ✔ WebView2: 129.0.2792.65
    ✔ MSVC: 
        - Visual Studio Build Tools 2019
        - Visual Studio Community 2022
    ✔ rustc: 1.80.1 (3f5fd8dd4 2024-08-06)
    ✔ cargo: 1.80.1 (376290515 2024-07-16)
    ✔ rustup: 1.27.1 (54dd3d00f 2024-04-24)
    ✔ Rust toolchain: stable-x86_64-pc-windows-msvc (environment override by RUSTUP_TOOLCHAIN)
    - node: 20.9.0
    - pnpm: 9.7.0
    - yarn: 1.22.19
    - npm: 10.8.2

[-] Packages
    - tauri 🦀: 2.0.0
    - tauri-build 🦀: 2.0.0
    - wry 🦀: 0.44.1
    - tao 🦀: 0.30.2
    - tauri-cli 🦀: 2.0.1
    - @tauri-apps/api : 2.0.1
    - @tauri-apps/cli : 2.0.1

[-] Plugins
    - tauri-plugin-store 🦀: 2.0.0
    - @tauri-apps/plugin-store : 2.0.0
    - tauri-plugin-single-instance 🦀: 2.0.0
    - @tauri-apps/plugin-single-instance : not installed!
    - tauri-plugin-shell 🦀: 2.0.0
    - @tauri-apps/plugin-shell : 2.0.0
    - tauri-plugin-updater 🦀: 2.0.1
    - @tauri-apps/plugin-updater : 2.0.0
    - tauri-plugin-autostart 🦀: 2.0.0
    - @tauri-apps/plugin-autostart : 2.0.0

[-] App
    - build-type: bundle
    - CSP: unset
    - frontendDist: ../dist
    - devUrl: http://localhost:1420/
    - framework: React
    - bundler: Vite
```

### Stack trace

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by kittuov**
The command present in update-task.xml will fail if there are any spaces in either the project name or "username" of the target machine. 

Fixed the command to properly escape such scenarios 

> More info about the bug is present in the issue #11216

### CODE REVIEW

File: crates/tauri-bundler/src/bundle/windows/msi/update-task.xml [hunk 1]
```diff
@@ -37,7 +37,7 @@
   <Actions Context="Author">
     <Exec>
       <Command>cmd.exe</Command>
-      <Arguments>/c "%SYSTEMROOT%\System32\msiexec.exe /i %TEMP%\\{{product_name}}.msi {{msiexec_args}} /promptrestart"</Arguments>
+      <Arguments>/c ^"%SYSTEMROOT%\System32\msiexec.exe /i "%TEMP%\\{{product_name}}.msi" {{msiexec_args}} /promptrestart^"</Arguments>
```
Comment (amrbashir) [inline crates/tauri-bundler/src/bundle/windows/msi/update-task.xml:40]: why are we escaping the first quotes here, can't we use a single quote `'` instead?

**Review by amrbashir — COMMENTED** [review]
Could you also add a change file in `.changes` directory?

File: crates/tauri-bundler/src/bundle/windows/msi/update-task.xml (hunk 1 above)
Comment (kittuov) [reply crates/tauri-bundler/src/bundle/windows/msi/update-task.xml:40]: In windows single and double quotes dont work similar to mac or linux. 

in windows, you escape only the outer quotations in this case`^"` . [source](https://ss64.com/nt/syntax-esc.html#:~:text=When%20ECHOing%20a%20string%20with%20an%20odd%20number%20(or%20just%20an%20unknown%20number)%20of%20embedded%20quotes%20through%20the%20pipeline%2C%20the%20outer%20quotes%20should%20be%20escaped%3A)

Comment (kittuov) [reply crates/tauri-bundler/src/bundle/windows/msi/update-task.xml:40]: image

Comment (kittuov) [reply crates/tauri-bundler/src/bundle/windows/msi/update-task.xml:40]: Just added change message. Let me know if that works

Comment (amrbashir) [reply crates/tauri-bundler/src/bundle/windows/msi/update-task.xml:40]: thanks for clarification

File: .changes/fix-elevated-update-cmd.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"tauri-bundler": patch:fix
```
Comment (amrbashir) [inline .changes/fix-elevated-update-cmd.md]:
```suggestion
"tauri-bundler": patch:bug
```

File: .changes/fix-elevated-update-cmd.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+---
+
+Fixed Command arguments for Windows -> msi -> elevatedUpdateTask. This works with spaces in project name on windows
```
Comment (amrbashir) [inline .changes/fix-elevated-update-cmd.md]:
```suggestion
On Windows, fixed command arguments for `bundle -> windows -> msi -> elevatedUpdateTask`. to work with spaces in `productName`
```

**Review by amrbashir — CHANGES_REQUESTED** [review]

**Review by amrbashir — APPROVED** [review]
Thank you

**Pull request merged**